src/libs/ContactUtils.ts

```diff
@@ -0,0 +1,51 @@
+import CONST from '@src/CONST';
+import type {PersonalDetails} from '@src/types/onyx';
+import type {DeviceContact, StringHolder} from './ContactImport/types';
+import {getUserToInviteContactOption} from './OptionsListUtils';
+import type {SearchOption} from './OptionsListUtils';
+import RandomAvatarUtils from './RandomAvatarUtils';
+
+function sortEmailObjects(emails?: StringHolder[]): string[] {
+    if (!emails?.length) {
+        return [];
+    }
+
+    const expensifyDomain = CONST.EMAIL.EXPENSIFY_EMAIL_DOMAIN.toLowerCase();
+
+    return emails
```

Comment:
NAB micro-optimization:

```ts
return emails
    .reduce((memo, email) => {
        if (email?.value) {
            memo.push(email);
        }
        return memo;
    }, [])
    .sort(...);
```

Reply:
Good microoptimizations, thanks!
Just to double check - did you test it with Hermes and not only on web? Because Hermes did a lot of work with array methods and they're pretty good, and when I tested it a few months ago the difference was negligible

Reply:
Pushed it, thanks again!